src/Components/OATPropertyEditor/JSONEditor.tsx

```diff
@@ -75,20 +75,54 @@ const JSONEditor = ({ dispatch, theme, state }: JSONEditorProps) => {
         });
     }
 
+    const onCancelClick = () => {
+        setContent(unSavedContent.current);
+        dispatch({ type: SET_OAT_DISABLED, payload: false });
+    };
+
+    const onSaveClick = async () => {
+        const newModel = isJsonStringValid(content);
+        const validJson = await parseModels([newModel]);
+        if (validJson) {
+            dispatch({
+                type: SET_OAT_PROPERTY_EDITOR_MODEL,
+                payload: newModel
+            });
+            unSavedContent.current = content;
+            dispatch({ type: SET_OAT_DISABLED, payload: false });
+        }
+    };
+
     return (
-        <Editor
-            defaultLanguage="json"
-            value={content}
-            onMount={onHandleEditorDidMount}
-            onChange={onHandleEditorChange}
-            theme={
-                themeToUse === 'dark' || theme === 'explorer'
-                    ? 'vs-dark'
-                    : themeToUse
-            }
-            beforeMount={setEditorThemes}
-            height={'95%'}
-        />
+        <>
+            {state.disabled && (
```

Comment:
Does this state property have the best name?

It seems weird to show the buttons when disabled == true. Would "modified" be a better name?